packages/devtools_app/lib/src/analytics/analytics.dart

```diff
@@ -214,6 +226,35 @@ void screen(
   );
 }
 
+void timing(
+  String screenName,
+  String timedOperation, {
+  @required Duration duration,
```

Comment:
Comment to clarify that duration should always refers to time DevTools took not time an app took
I would suggest creating a DevToolsScreen metrics objects that groups 
cpuSampleCount, cpuStackDepth, and traceEventCount. Those are very performance page specific so it is a bit surprising to see them in this method. Keeping them separate will make the important parts of this API clearer and make it easier if you chose to refactor how those metrics are exposed. Internally it could be just a key value pair map or something.
I would also suggest a timing api that is more similar to what dart:developer does. So for example:
```dart
AnalyticsTimeline.timeSync("screenName", "someTimedOperation", () {
  doSomething();
}, ScreenMetrics(cpuSampleCount: 1000, inspectorTreeSize: 100000)
);```
https://api.dart.cn/stable/2.13.4/dart-developer/Timeline-class.html
I'm less sure about the async version of the API as the one in dart:developer seems like it was written before Dart supported async. Something like this might make sense:
```dart
AnalyticsTimeline.timeAsync("screenName", "someTimedOperation", someFutureThatCompleteAtTheEndTime, ScreenMetrics(cpuSampleCount: 1000, inspectorTreeSize: 100000)
);```

Reply:
did some significant refactoring - PTAL